app/api/route.js

```diff
@@ -0,0 +1,96 @@
+import Author from "@/models/authors";
+import DataLoader from "dataloader";
+import { ApolloServer } from "@apollo/server";
+import { startServerAndCreateNextHandler } from "@as-integrations/next";
+import { gql } from "graphql-tag";
+const Book = require("../../models/books");
+const { Op } = require("sequelize");
+const typeDefs = gql`
+  type Book {
+    title: String!
+    description: String!
+    publishedAt: String!
+    author: Author!
+  }
+
+  type Author {
+    name: String!
+    biography: String!
+    books: [Book!]!
+    dateOfBirth: String!
+  }
+
+  type Query {
+    book: Book
+    books: [Book!]!
+    author: Author
+    authors: [Author!]!
+  }
+`;
+
+const batchAuthors = new DataLoader(async (author_ids) => {
+  const authorsList = await Author.findAll({
+    where: {
+      id: {
+        [Op.in]: author_ids,
+      },
+    },
+  });
+  // Dataloader expects you to return a list with the results ordered just like the list in the arguments were
+  // Since the database might return the results in a different order the following code sorts the results accordingly
+  const idToAuthor = authorsList.reduce((mapping, author) => {
+    mapping[author.id] = author;
+    return mapping;
+  }, {});
+  return author_ids.map((id) => idToAuthor[id]);
```

Comment:
@shubhamtibra Please try to use lodash here, we use it thoroughly in our codebase so it will be helpful to get some practise!